### DISCUSSION TITLE
Lesson6: Error: Failed to create wallet from private key. Private key is invalid hex: Odd number of digits

### DISCUSSION THREAD

**Discussion by AiULe** [Cyfrin/foundry-full-course-cu#1222] [Q&A]
Hello everyone，I met a problem，When I run the following command
`forge script script/DeploySimpleStorage.s.sol --rpc-url http://127.0.0.1:8545 --broadcast --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2f`
terminal show following Error:
`Failed to create wallet from private key. Private key is invalid hex: Odd number of digits`
Can you take a look at it for me? thank you very much

**Comment by AiULe** [comment]
Hold on ,  I use `.env` ，Configuration `$PRIVATE_KEY` and `$RPC_URL` , I run the following command:
`forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL 
--broadcast --private-key $PRIVATE_KEY`
terminal show Success
Is this a bug?

**Comment by xyz899** [comment]
Your private key is the first of the provided anvils keys, this is the corrected version : `0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80`, yours is missing `f80 at the end` remember Private keys 32 bytes so 64 characters that's why the error
